boot/bootutil/zephyr/src/boot_request_retention.c

```diff
@@ -0,0 +1,341 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ */
+#include <bootutil/boot_request.h>
+
+#include <zephyr/retention/retention.h>
+#include "bootutil/bootutil_log.h"
+
+/** Special value of image number, indicating a request to the bootloader. */
+#define BOOT_REQUEST_IMG_BOOTLOADER 0xFF
+
+/** Helper value, indicating how many images are supported. */
+#define BOOT_REQUEST_IMAGES 2
+
+/** Number of requests per image. */
+#define BOOT_REQUEST_PER_IMAGE 2
```

Comment:
replace with `BOOT_IMAGE_NUMBER` for number of images, `BOOT_NUM_SLOTS` for number of slots, since they already exist

Reply:
Replaced with: `CONFIG_UPDATEABLE_IMAGE_NUMBER` since `BOOT_IMAGE_NUMBER` is not available on the application side.
The `BOOT_REQUEST_PER_IMAGE` is equal to the number of slots only accidentally - it represents the number of different request types that can be stored per image.